xdsl/dialects/arm_neon.py

```diff
@@ -62,9 +74,124 @@ def infinite_register_prefix(cls):
 V30 = NEONRegisterType.from_name("v30")
 V31 = NEONRegisterType.from_name("v31")
 
+
+class ARMNEONInstruction(ARMInstruction, ABC):
+    """
+    Base class for operations that can be a part of x86 assembly printing. Must
+    represent an instruction in the x86 instruction set.
+    The name of the operation will be used as the x86 assembly instruction name.
+    """
+
+    arrangement = attr_def(StringAttr)
+    """
+    An optional comment that will be printed along with the instruction.
+    """
+
+    def assembly_line(self) -> str | None:
+        # default assembly code generator
+        instruction_name = self.assembly_instruction_name()
+        arg_str = ", ".join(
+            f"{assembly_arg_str(arg)}.{self.arrangement.data}"
+            for arg in self.assembly_line_args()
+            if arg is not None
+        )
+        return AssemblyPrinter.assembly_line(instruction_name, arg_str, self.comment)
+
+
+@irdl_op_definition
+class GetRegisterOp(ARMOperation):
```

Comment:
why add this?

Reply:
because I have the neon register type in arm_neon (so I need an arm_neon.get_register op for the filecheck)

Reply:
"test.op" should do the trick for the time being, no?

Reply:
Ah it's for assembly printing